src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

```diff
@@ -312,15 +313,29 @@ private AIState deliver()
                     final ItemStack insertionResultStack = buildingToDeliver.forceTransferStack(stack, world);
                     if (!InventoryUtils.isItemStackEmpty(insertionResultStack))
                     {
-                        chatSpamFilter.talkWithoutSpam("com.minecolonies.coremod.job.deliveryman.workerChestFull"
-                          , new TextComponentString(" :" + buildingToDeliver.getSchematicName()));
+                        if (ItemStack.areItemStacksEqual(insertionResultStack, stack))
+                        {
+                            final AbstractBuildingWorker workerBuilding = (AbstractBuildingWorker) buildingToDeliver;
+                            //same stack, we could not deliver ?
+                            if (workerBuilding.getWorker() != null)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)